recsyslearn/segmentations.py

```diff
@@ -175,10 +175,43 @@ def segment(self, dataset: pd.DataFrame, proportion=None, min_interaction: int =
         user_groups.loc[:, 'count'] = user_groups.loc[:, 'count'].apply(lambda x: x + np.random.choice(list(range(10))))
         user_groups = user_groups.sort_values('count', ascending=False)
         user_groups.loc[:, 'count'] = np.arange(user_groups.shape[0]) + 1
-        threshold = round(user_groups.shape[0] * proportion[0])\
+        threshold = round(user_groups.shape[0] * proportion[0]) \
             if round(user_groups.shape[0] * proportion[0]) > 0 else 1
 
         user_groups.loc[user_groups['count'] <= threshold, 'group'] = '1'
         user_groups.loc[user_groups['count'] > threshold, 'group'] = '2'
 
         return user_groups[['user', 'group']]
+
+
+class DiscreteFeatureSegmentation(Segmentation):
+    """
+    Segmentation of entities (users or items) according to one of
+    their features (e.g., gender for users or genre for items)
+    """
+
+    def segment(self, feature: pd.DataFrame, group: str = 'item', ) -> pd.DataFrame:
+        """
+        Segmentation of users/items based on one of their features.
+        Before assigning the group, the nans are given a -1 value by default.
+        Make sure that this is not one of the feature values, already.
+
+        Parameters
+        ----------
+        feature : pd.DataFrame
+            The feature dataframe in form of [id, feature] storing the categorical feature
+            to be used for grouping.
+
+        group : str
```

Comment:
We are not using this variable. Could we remove it?

Reply:
You are right, fine with that! I'm on it.